File: app/src/main/java/com/hiddenramblings/tagmo/widget/Toasty.java
package com.hiddenramblings.tagmo.widget;

import android.app.Activity;
import android.widget.Toast;

import androidx.appcompat.app.AlertDialog;

import com.hiddenramblings.tagmo.R;

import java.lang.ref.WeakReference;

public class Toasty {

    private final WeakReference<Activity> mActivity;

    public Toasty(Activity activity) {
        mActivity = new WeakReference<>(activity);
    }

    private void show(int msgRes, int length) {
        mActivity.get().runOnUiThread(() -> Toast.makeText(
                mActivity.get(), msgRes, length).show());
    }

    @SuppressWarnings("unused")
    public void Long(int msgRes) {
        show(msgRes, Toast.LENGTH_LONG);
    }

    public void Short(int msgRes) {
        show(msgRes, Toast.LENGTH_SHORT);
    }

    private void show(String msg, int length) {
        mActivity.get().runOnUiThread(() -> Toast.makeText(
                mActivity.get(), msg, length).show());
    }

    public void Long(String msg) {
        show(msg, Toast.LENGTH_LONG);
    }

    public void Short(String msg) {
        show(msg, Toast.LENGTH_SHORT);
    }

    public void Dialog(String msg) {
        mActivity.get().runOnUiThread(() -> new AlertDialog.Builder(mActivity.get())
                .setMessage(msg).setPositiveButton(R.string.close, null).show());
    }

    public void Dialog(int msgRes) {
        mActivity.get().runOnUiThread(() -> new AlertDialog.Builder(mActivity.get())
                .setMessage(msgRes).setPositiveButton(R.string.close, null).show());
    }
}

File: app/src/main/java/com/hiddenramblings/tagmo/widget/Toasty.kt
package com.hiddenramblings.tagmo.widget

import android.content.Context
import android.os.Handler
import android.os.Looper
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import com.hiddenramblings.tagmo.R


class Toasty(context: Context) {
    private val mContext: Context

    init {
        mContext = context
    }

    private fun show(msgRes: Int, length: Int) {
        Handler(Looper.getMainLooper()).post {
            Toast.makeText(mContext, msgRes, length).show()
        }
    }

    fun Long(msgRes: Int) {
        show(msgRes, Toast.LENGTH_LONG)
    }

    fun Short(msgRes: Int) {
        show(msgRes, Toast.LENGTH_SHORT)
    }

    private fun show(msg: String, length: Int) {
        Handler(Looper.getMainLooper()).post {
            Toast.makeText(mContext, msg, length).show()
        }
    }

    fun Long(msg: String) {
        show(msg, Toast.LENGTH_LONG)
    }

    fun Short(msg: String) {
        show(msg, Toast.LENGTH_SHORT)
    }

    fun Dialog(msg: String?) {
        Handler(Looper.getMainLooper()).post {
            AlertDialog.Builder(mContext)
                .setMessage(msg)
                .setPositiveButton(R.string.close, null)
                .show()
        }
    }

    fun Dialog(msgRes: Int) {
        Handler(Looper.getMainLooper()).post {
            AlertDialog.Builder(mContext)
                .setMessage(msgRes)
                .setPositiveButton(R.string.close, null)
                .show()
        }
    }
}